app/src/main/java/com/poohcom1/playalong/ui/scenes/SongController.kt

```diff
@@ -21,21 +19,32 @@ import com.yausername.youtubedl_android.mapper.VideoInfo
 fun SongController(info: VideoInfo, modifier: Modifier = Modifier) {
     var start: Float by remember { mutableStateOf(0f) }
     var end: Float by remember { mutableStateOf(info.duration.toFloat()) }
+    var currentPosition by remember { mutableStateOf(0L) }
+
+    val videoURL = info.url
 
     var loopRange: LongRange by remember { mutableStateOf(0L..info.duration * 1000) }
 
     Column(
         Modifier.padding(16.dp),
         horizontalAlignment = Alignment.CenterHorizontally
     ) {
-        VideoPlayer(info.url!!, loopRange, 120f)
+        if (videoURL != null) {
+            VideoPlayer(videoURL, loopRange, 120f, remember { mutableStateOf(currentPosition) })
```

Comment:
Fix this as well once you correct the parameter type. Wrapping a state into another state is not a good idea and can break reactivity